services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/switches/SwitchRulesSpec.groovy

```diff
@@ -421,6 +427,56 @@ class SwitchRulesSpec extends BaseSpecification {
         flowHelper.deleteFlow(flow.id)
     }
 
+    def "Traffic counters in ingress rule are reset on flow rerouting"() {
+        given: "Two active switches and two possible flow paths at least"
+        def switches = topology.getActiveSwitches()
+        List<List<PathNode>> possibleFlowPaths = []
+        def (Switch srcSwitch, Switch dstSwitch) = [switches, switches].combinations()
+                .findAll { src, dst -> src != dst }.find { Switch src, Switch dst ->
+            possibleFlowPaths = database.getPaths(src.dpId, dst.dpId)*.path.sort { it.size() }
+            possibleFlowPaths.size() > 1
+        } ?: assumeTrue("No suiting switches found", false)
+
+        and: "Create a flow going through these switches"
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flowHelper.addFlow(flow)
+
+        when: "Ping the flow"
+        def response = northbound.pingFlow(flow.id, new PingInput())
+
+        then: "Operation is successful"
+        response.forward.pingSuccess
+        response.reverse.pingSuccess
+
+        and: "Traffic counters in ingress rule on source and destination switches represent packets movement"
+        checkTrafficCountersInRules(flow.source)
+        checkTrafficCountersInRules(flow.destination)
+
+        when: "Break a flow ISL (bring switch port down) to cause flow rerouting"
+        def flowPath = PathHelper.convert(northbound.getFlowPath(flow.id))
+        def flowIsls = pathHelper.getInvolvedIsls(flowPath)
+
+        Set<Isl> altFlowIsls = []
+        possibleFlowPaths.findAll { it != flowPath }.each { altFlowIsls.addAll(pathHelper.getInvolvedIsls(it)) }
+
+        def islToFail = flowIsls.find { !(it in altFlowIsls) && !(islUtils.reverseIsl(it) in altFlowIsls) }
+        northbound.portDown(islToFail.srcSwitch.dpId, islToFail.srcPort)
+
+        then: "The flow was rerouted after reroute timeout"
+        Wrappers.wait(rerouteDelay + WAIT_OFFSET) {
+            assert northbound.getFlowStatus(flow.id).status == FlowState.UP
+            assert PathHelper.convert(northbound.getFlowPath(flow.id)) != flowPath
+        }
+
+        and: "Traffic counters in ingress rule on source and destination switches are reset"
+        checkTrafficCountersInRules(flow.source, false)
+        checkTrafficCountersInRules(flow.destination, false)
+
+        and: "Revive the ISL back (bring switch port up) and delete the flow"
+        northbound.portUp(islToFail.srcSwitch.dpId, islToFail.srcPort)
```

Comment:
You should wait for ISL to become discovered

Reply:
Thanks, fixed.